runtime/lua/vim/lsp/health.lua

```diff
@@ -33,16 +33,25 @@ local function check_active_clients()
   local clients = vim.lsp.get_clients()
   if next(clients) then
     for _, client in pairs(clients) do
-      local attached_to = table.concat(vim.tbl_keys(client.attached_buffers or {}), ',')
-      report_info(
+      local cmd ---@type string
+      if type(client.config.cmd) == 'table' then
+        cmd = table.concat(client.config.cmd --[[@as table]], ' ')
+      elseif type(client.config.cmd) == 'function' then
+        cmd = tostring(client.config.cmd)
+      end
+      report_info(table.concat({
+        client.name,
+        string.format('  ID: %d', client.id),
+        string.format('  Root directory: %s', vim.fn.fnamemodify(client.root_dir, ':~')),
+        string.format('  Command: %s', cmd),
+        string.format('  Offset encoding: %s', string.upper(client.offset_encoding)),
```

Comment:
To make the representation a bit more compact I'd tend to:

- Remove the flags - they're usually not very important
- Move the id the into the same line as the name `clang (id: 1)`
- Only show offset encoding if it is either diverging from the default, or if there are mixed encodings across clients.

But also okay as is.

Reply:
>Only show offset encoding if it is either diverging from the default, or if there are mixed encodings across clients.

I think this would be more trouble than it's worth. You and I know that the default offset encoding is UTF-16, but the "average" user may not. So if one client says `Offset encoding: UTF-8` and another client doesn't say anything, I suspect the reaction will be "why is offset encoding not listed here?" rather than "oh this must just be using the default value".

Reply:
But maybe we can just remove it completely. Not sure how valuable it is as far as debugging information goes. I think it could be useful for someone to verify that their configuration is working as expected though ("did I set the `positionEncodings` capability correctly?")

Reply:
> "did I set the positionEncodings capability correctly?"

But that applies to any capability, no?

>  Which makes me wonder if we should be labeling this "Position encoding" rather than "Offset encoding"?

yes

> So if one client says Offset encoding: UTF-8 and another client doesn't say anything, I suspect the reaction will be "why is offset encoding not listed here?" rather than "oh this must just be using the default value".

Fair point. How about showing it everywhere if there are mixed encodings, and otherwise nowhere? (Given that currently mixed encodings are known to cause problems with some of the functions)

Reply:
>Fair point. How about showing it everywhere if there are mixed encodings, and otherwise nowhere? (Given that currently mixed encodings are known to cause problems with some of the functions)

Ok so I went ahead and just made a separate health check for this. We already show a warning when this happens, but now we have it in `:healthcheck` too.

```
vim.lsp: Position Encodings ~
- WARNING Found buffers attached to multiple clients with different position encodings.
  - Buffer 2: UTF-16 (client id(s): 2), UTF-8 (client id(s): 1)
  - ADVICE:
    - Use the positionEncodings client capability to ensure all clients use the same position encoding
```